nixos/doc/manual/development/option-types.section.md

```diff
@@ -317,6 +317,24 @@ If the you're interested in can be distinguished without a label, you may simpli
 :   An attribute set containing one attribute, whose name must be picked from
     the attribute set (`attr1`, etc) and whose value consists of definitions that are valid for the corresponding option (`option1`, etc).
 
+    ```nix
+    mkOption {
+      type = attrTag {
+        foo = mkOption {
+          type = bool;
+        };
+        bar = mkOption {
+          type = int;
+        };
+      };
+    }
+    ```
+
+    Can be thought of as equivalent to the following pseudotype
+
+    ```pseudocode
+    { foo :: Bool } | { bar :: Int }
```

Comment:
This is doubly confusing, because you're conflating pseudo code and some habit of more concise type notations. I'd just leave that out. The illustration of the prose should be enough to get the idea.